app/src/main/kotlin/com/hippo/ehviewer/ui/settings/DownloadScreen.kt

```diff
@@ -162,7 +164,14 @@ fun DownloadScreen(navigator: DestinationsNavigator) {
                 title = stringResource(id = R.string.settings_download_reload_metadata),
                 summary = stringResource(id = R.string.settings_download_reload_metadata_summary),
             ) {
-                DownloadManager.downloadInfoList.filter { it.state == DownloadInfo.STATE_FINISH }.apply {
+                fun DownloadInfo.isStable(): Boolean {
+                    val downloadTime = downloadDir?.findFile(COMIC_INFO_FILE)?.lastModified() ?: return false
+                    // stable after 30 days
+                    val stableTime = postedTime?.plus(30L * 24L * 60L * 60L * 1000L) ?: return false
+                    return downloadTime > stableTime
+                }
+
+                DownloadManager.downloadInfoList.filter { it.state == DownloadInfo.STATE_FINISH && !it.isStable() }.apply {
```

Comment:
Run it in parallel and limit max concurrency.

Reply:
Why we need to limit concurrency for local file read/write?

Reply:
I was afraid it would take too much resources.
But after reconsidering, I think it's fine to not limit it since it's running foreground.